tests/integration/relations/opensearch_provider/application-charm/src/charm.py

```diff
@@ -35,37 +35,39 @@ def __init__(self, *args):
 
         # Events related to the first database that is requested
         # (these events are defined in the database requires charm library).
-        database_name = f'{self.app.name.replace("-", "_")}_first_database'
+        index_name = f'{self.app.name.replace("-", "_")}_first_opensearch'
 
         permissive_roles = json.dumps({"roles": ["all_access"]})
-        self.first_database = DatabaseRequires(
-            self, "first-database", database_name, permissive_roles
+        self.first_opensearch = OpenSearchRequires(
+            self, "first-index", index_name, permissive_roles
         )
         self.framework.observe(
-            self.first_database.on.database_created, self._on_first_database_created
+            self.first_opensearch.on.index_created, self._on_first_opensearch_created
         )
         self.framework.observe(
-            self.first_database.on.endpoints_changed, self._on_first_database_endpoints_changed
+            self.first_opensearch.on.endpoints_changed, self._on_first_opensearch_endpoints_changed
         )
 
         # Events related to the second database that is requested
         # (these events are defined in the database requires charm library).
-        database_name = f'{self.app.name.replace("-", "_")}_second_database'
-        # TODO change this to use new permissions
+        index_name = f'{self.app.name.replace("-", "_")}_second_opensearch'
+        # TODO change this to include only permissions and action groups, and verify that we can
+        # create roles when necessary.
         roles = {
             "roles": ["readall"],
             "permissions": ["TODO find some permissions", ""],
             "action_groups": ["TODO find some action groups", ""],
```

Comment:
Are we still passing `permissions` and `action_groups`? thought we had the `extra_user_roles` passed as an inline string i.e `admin`?

Reply:
That's planned for a future PR, but yes